public/src/components/dashboard/SubmissionsTable.tsx

```diff
@@ -27,66 +26,75 @@ const SubmissionsTable = () => {
+            */
+            if (date.getFullYear() !== now.getFullYear() || date.getMonth() > now.getMonth() + 1) {
                 return
             }
-            // Submissions are indexed by the assignment order - 1.
-            const submission = submissions.find(sub => sub.AssignmentID === assignment.ID) ?? create(SubmissionSchema)
-            const status = getStatusByUser(submission, state.self.ID)
-            if (!isApproved(status) && assignment.deadline) {
-                const deadline = timeFormatter(assignment.deadline)
-                if (deadline.daysUntil > 3 && submission.score >= assignment.scoreLimit) {
-                    deadline.className = "table-success"
-                }
-                if (!deadline.message) {
-                    return
-                }
-                const course = state.courses.find(c => c.ID === courseID)
-                table.push(
-                    <tr key={assignment.ID.toString()} className={`clickable-row ${deadline.className}`}
-                        onClick={() => history.push(`/course/${courseID}/lab/${assignment.ID}`)}>
-                        <th scope="row">{course?.code}</th>
-                        <td>
-                            {assignment.name}
-                            {assignment.isGroupLab ?
-                                <span className="badge ml-2 float-right"><i className={Icon.GROUP} title="Group Assignment" /></span> : null}
-                        </td>
-                        <td><ProgressBar courseID={courseID.toString()} submission={submission} type={Progress.OVERVIEW} /></td>
-                        <td>{getFormattedTime(assignment.deadline, true)}</td>
-                        <td>{deadline.message ? deadline.message : '--'}</td>
-                        <td className={SubmissionStatus[status]}>
-                            {assignmentStatusText(assignment, submission, status)}
-                        </td>
-                    </tr>
-                )
-            }
-        })
-        return table
-    }
+
+            const deadline = timeFormatter(date, submission.score >= assignment.scoreLimit)
+            const course = state.courses.find(c => c.ID === courseID)
+            table.push(
+                <tr key={assignment.ID.toString()} className={`clickable-row ${deadline.className}`}
+                    onClick={() => history.push(`/course/${courseID}/lab/${assignment.ID}`)}>
+                    <th scope="row">{course?.code}</th>
+                    <td>
+                        {assignment.name}
+                        {assignment.isGroupLab ?
+                            <span className="badge ml-2 float-right"><i className={Icon.GROUP} title="Group Assignment" /></span> : null}
+                    </td>
+                    <td><ProgressBar courseID={courseID.toString()} submission={submission} type={Progress.OVERVIEW} /></td>
+                    <td>{getFormattedTime(assignment.deadline, true)}</td>
```

Comment:
Perhaps this should be called internally in `deadlineFormatter` and assigned to a field `{deadline.time}`.